staging/src/k8s.io/apimachinery/pkg/util/jsonmergepatch/patch.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jsonmergepatch
+
+import (
+	"fmt"
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/util/json"
+	"github.com/evanphx/json-patch"
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
```

Comment:
it's weird for jsonmergepatch to depend on strategicpatch. ok as a follow-up, but moving the precondition funcs, errors, conflict and toYAML helpers to a neutral package would make sense to me